src/segmentation.rs

```diff
@@ -28,6 +28,12 @@ pub fn segmentation_optimize<T: Pixel>(
   // showed this to be the optimal one.
   const TEMPORAL_RDO_QI_DELTA: i16 = 21;
 
+  // Avoid going into lossless mode by never bringing qidx below 1.
```

Comment:
I don't know where the rate-control adjusted frame qidx is stored, but at this point I believe that info is available (i.e. frame level rate control is finished). If so, can we check what block qidx will with each of three qi_delta from three seg id and disable them by setting segmentation.features[ ] inside the for loop below?
I simply hope, by doing so, we are able to block the use of qidx <= 1 from the origin?

Reply:
The segmentation adjusted qidx is fetched by the get_qidx() function. We can force it in the loop below instead. The only downside is that we'll end up searching sidx 1 twice if we're at the limit and doing segment rdo.

Reply:
Hmm, I think I didn't get what you mean. I double checked that this function segmentation_optimize(), where this comment box is located, is called by encode_frame(), only once for a frame. Meanwhile, get_qidx() is called in the beginning of encode_tx_block() to fetch the qidx to use with quantizatioin and dequant.
One issue I now expect is, quant_rdo seems always assume that there are three qidx available (accessed by three seg id) then assign one to the block after quant_rdo is done!

Reply:
> The only downside is that we'll end up searching sidx 1 twice if we're at the limit and doing segment rdo.

"The only downside is that we'll end up searching sidx 1 twice if we're at the limit and doing segment rdo.", this means seg id 1 and 2 get merged since seg id 2 introduce qidx <= 1, right?

Reply:
segmentation_optimize() only gets called for intra frames, so those values only change every keyframe. My limits in that function prevent it using lossless 99% of the time. But there's still a small chance that due to base_q_idx changing, segment 2 could still enter lossless, so we still need to check that one as well.

Yes, everything now assumes there are exactly 3 segment ids, and 3 qidx available. It's kind of intentionally hardcoded right now.

Reply:
I see. I didn't acknowledge that the function is only called for keyframe. Then I guess that, w/o your PR here, the 1% chance is when certain inter frame to have qinx <= 21 (i.e. 22 - 21 = 0).
Please go ahead merging.